src/oracle.c

```diff
@@ -1190,26 +1185,28 @@ main(int argc, char **argv)
 		break;
 
 	case ACTION_UNSEAL:
-		if (tpm2key_fmt) {
-			end_arguments(argc, argv);
-			break;
+		action_flags = target_platform_unseal_flags(target);
+		if (action_flags & PLATFORM_OPTIONAL_PCR_POLICY) {
+			if (opt_rsa_public_key == NULL && opt_pcr_policy)
+				usage(1, "You need to specify the --public-key option when unsealing using an authorized policy\n");
+			if (opt_pcr_policy == NULL && opt_rsa_public_key)
+				usage(1, "You need to specify the --pcr-policy option when unsealing using an authorized policy\n");
 		}
 
-		if (opt_rsa_public_key == NULL && opt_pcr_policy)
-			usage(1, "You need to specify the --public-key option when unsealing using an authorized policy\n");
-		if (opt_pcr_policy == NULL && opt_rsa_public_key)
-			usage(1, "You need to specify the --pcr-policy option when unsealing using an authorized policy\n");
-		pcr_selection = get_pcr_selection_argument(argc, argv, opt_algo);
+		if ((action_flags & PLATFORM_NEED_INPUT_FILE) && !opt_input)
+			usage(1, "You need to specify an input file via --input when unsealing a secret");
+		if ((action_flags & PLATFORM_NEED_OUTPUT_FILE) && !opt_output)
+			usage(1, "You need to specify an output file via --output when unsealing a secret");
+		if (action_flags & PLATFORM_NEED_PCR_SELECTION)
+			pcr_selection = get_pcr_selection_argument(argc, argv, opt_algo);
 		end_arguments(argc, argv);
 		break;
 
 	case ACTION_SIGN:
 		if (opt_rsa_private_key == NULL)
 			usage(1, "You need to specify the --private-key option when signing a policy\n");
-		if (systemd_json && opt_output == NULL)
-			usage(1, "You need to specify the --output option when signing a systemd policy\n");
-		if (tpm2key_fmt && opt_input == NULL)
-			usage(1, "You need to specify the --input option when signing a policy into a TPM 2.0 Key file\n");
+		if (opt_output == NULL)
```

Comment:
there is not an associated `action_flags & PLATFORM_NEED_OUTPUT_FILE` for `ACTION_SIGN`?

Reply:
Yeah, right now I'm doing the action_flags dance only for unseal, because there is so much variation there. I'll probably do something similar once we've agreed on how we want to implement the signing action across all target platforms.